tracing/examples/sloggish/main.rs

```diff
@@ -21,31 +21,32 @@ fn main() {
     let subscriber = SloggishSubscriber::new(2);
     tracing::subscriber::set_global_default(subscriber).unwrap();
 
-    span!(Level::TRACE, "", version = &field::display(5.0)).in_scope(|| {
-        span!(Level::TRACE, "server", host = "localhost", port = 8080).in_scope(|| {
-            info!("starting");
-            info!("listening");
-            let peer1 = span!(Level::TRACE, "conn", peer_addr = "82.9.9.9", port = 42381);
-            peer1.in_scope(|| {
-                debug!("connected");
-                debug!({ length = 2 }, "message received");
-            });
-            let peer2 = span!(Level::TRACE, "conn", peer_addr = "8.8.8.8", port = 18230);
-            peer2.in_scope(|| {
-                debug!("connected");
-            });
-            peer1.in_scope(|| {
-                warn!({ algo = "xor" }, "weak encryption requested");
-                debug!({ length = 8 }, "response sent");
-                debug!("disconnected");
-            });
-            peer2.in_scope(|| {
-                debug!({ length = 5 }, "message received");
-                debug!({ length = 8 }, "response sent");
-                debug!("disconnected");
-            });
-            warn!("internal error");
-            info!("exit");
-        })
+    let app_span = span!(Level::TRACE, "", version = %5.0);
```

Comment:
is the % here required? I'm assuming that display is the default. If its already defaulting to display I'm wondering why you'd ever want to use %

Reply:
The default is record types that implement `Value` with their `Value` implementation. Otherwise, for types which don't have `Value` implementations, it is necessary to specify how they are to be recorded. `f64` and `f32` do not implement `Value`, as they are not primitive value types that the `Record` trait knows about.